src/matches.rs

```diff
@@ -402,7 +402,7 @@ fn rewrite_match_body(
         let arrow_snippet = context.snippet(arrow_span).trim();
         // search for the arrow starting from the end of the snippet since there may be a match
         // expression within the guard
-        let arrow_index = arrow_snippet.rfind("=>").unwrap();
+        let arrow_index = arrow_snippet.find_last_uncommented("=>").unwrap();
```

Comment:
I've been thinking a little about this and I think we should also add a [version gate](https://github.com/rust-lang/rustfmt/blob/master/Contributing.md#version-gate-formatting-changes) for this fix. If `version::One` is set we'll go with the old `arrow_snippet.rfind("=>")` code, otherwise we'll use the updated, and more accurate `arrow_snippet.find_last_uncommented("=>")`